fullstack-cert/python-projects/bisection-method-lab/user-stories.md

```diff
@@ -0,0 +1,32 @@
+The bisection method is a technique for finding the roots of a real-valued function. It works by narrowing down an interval where the square root lies until it converges to a value within a specified tolerance.
+
+For example, if the tolerance is `0.01`, the bisection method will keep halving the interval until the difference between the upper and lower bounds is less than or equal to `0.01`.
+
+In this lab, you will create a function that uses the bisection method to find the square root of a number.
+
+**User stories**
+
+1. You should define a function named `square_root_bisection` that takes three parameters:
+   - The number for which you want to find the square root.
+   - Tolerance being the acceptable error margin for the result. You should set a default tolerance value.
+   - The maximum number of iterations to perform.
+
+2. Within the `square_root_bisection` function, you should first check the following conditions:
+    - If the number is negative, you should raise a `ValueError` with the message `"Square root of negative number is not defined in real numbers"`.
+    - If the number is `1`, you should return the message `"The square root of 1 is 1"`.
+    - If the number is `0`, you should return the message `"The square root of 0 is 0"`.
```

Comment:
```suggestion
    - If the number is `1`, you should print the message `The square root of 1 is 1`.
    - If the number is `0`, you should print the message `The square root of 0 is 0`.
```
I'm not sure if this is necessary. Maybe we should just tell them to return `The square root of <num> is <root>` without specifying that they should handle `1` and `0` differently.